### PR TITLE
Added rigid body axis locks

### PR DESCRIPTION
**Pull request by SergeyAMZN**
Allows to lock motion along/around a specific axis

### CODE REVIEW

**Review by hultonha — APPROVED** [review]
LGTM! 😄 Thanks!

File: Gems/PhysX/Code/Source/EditorRigidBodyComponent.cpp [hunk 1]
```diff
@@ -216,7 +216,19 @@ namespace PhysX
...
+                        ->Attribute(AZ::Edit::Attributes::Min, 0.f)
+                        ->Attribute(AZ::Edit::Attributes::Max, 1.f)
+                        ->Attribute(AZ::Edit::Attributes::Step, 1.f)
+
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &RigidBodyConfiguration::m_angularAxisLock,
+                        "Angular Axis Lock", "Lock rotation around a specified axis (0|1)")
+                        ->Attribute(AZ::Edit::Attributes::Min, 0.f)
+                        ->Attribute(AZ::Edit::Attributes::Max, 1.f)
+                        ->Attribute(AZ::Edit::Attributes::Step, 1.f)
+                        ;
```
Comment (amzn-sean) [inline Gems/PhysX/Code/Source/EditorRigidBodyComponent.cpp:231]: What does the UX look like for this?

Is there a reason we didn't make a AxisLock struct that reflects 3 bools to edit contexts. then here have 1 of these structs reflected for each lock here?

Comment (ibtehajn) [reply Gems/PhysX/Code/Source/EditorRigidBodyComponent.cpp:231]: I was working on the same thing and implemented it with a list of bool flags in a collapsible group. Granted, not the best UX (but I was after a quick fix). 
image

File: Gems/PhysX/Code/Source/RigidBody.cpp [hunk 2]
```diff
@@ -83,6 +83,8 @@ namespace PhysX
             {
                 SetUserData(configuration.m_customUserData);
             }
+
+            SetAxisLockFlags(configuration.m_linearAxisLock, configuration.m_angularAxisLock);
```
Comment (ibtehajn) [inline Gems/PhysX/Code/Source/RigidBody.cpp:87]: Given that the values (0/1) for the locking properties are coming from the `RigidBodyConfiguration` object, would it not make more sense to move this to `PxActorFactories::CreatePxRigidBody`? That's where I put the logic for my quick (hacky) fix. I also made the code that upgrades older serialized versions add appropriate defaults (which may not actually be required) to make it safe to move this to `PxActorFactories::CreatePxRigidBody`.

File: Gems/PhysX/Code/Source/EditorRigidBodyComponent.cpp (hunk 1 above)
Comment (SergeyAMZN) [reply Gems/PhysX/Code/Source/EditorRigidBodyComponent.cpp:231]: image
here the picture. 
Later when we get VectorBool3 working, the numbers would be replaced with checkboxes.
I think there's another PR making it a list of checkboxes, so see if that one is preferable

Comment (amzn-sean) [reply Gems/PhysX/Code/Source/EditorRigidBodyComponent.cpp:231]: the list of check boxes i think is a better UX.
if i had to choose between the vector of floats (that are not really floats) and checkboxes.

**Comment by kberg-amzn** [comment]
Seems like a pretty useful feature. I would be okay with something like this going in with a suboptimal UI *as long as* there was a guaranteed resolution date for an improved UI prior to general availability.

**Comment by amzn-sean** [comment]
> 
> 
> Seems like a pretty useful feature. I would be okay with something like this going in with a suboptimal UI _as long as_ there was a guaranteed resolution date for an improved UI prior to general availability.
@kberg-amzn 
Someone else made a change that took a slightly better UX approach (still suboptimal but better) so this PR was closed for that one instead. Not sure the new PR has been opened yet.

**Comment by hultonha** [comment]
@kberg-amzn @amzn-sean - this is the updated PR for reference - https://github.com/o3de/o3de/pull/2528

**Pull request closed without merging**